frontends/concrete-python/examples/prime-match/prime-match-sh.py

```diff
@@ -0,0 +1,113 @@
+import random
+import time
+
+import numpy as np
+
+from concrete import fhe
+
+NUMBER_OF_SYMBOLS = 10
+
+MINIMUM_ORDER_QUANTITY = 1
+MAXIMUM_ORDER_QUANTITY = 50
+
+assert 0 < MINIMUM_ORDER_QUANTITY < MAXIMUM_ORDER_QUANTITY
+
+
+def prime_match(
+    bank_order_quantities,
+    client_order_quantities,
+):
+    with fhe.tag("calculating-matching-order-quantity"):
+        return np.minimum(bank_order_quantities, client_order_quantities)
+
+
+inputset = [
+    (
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY, size=(NUMBER_OF_SYMBOLS * 2,)
+        ),
+        np.random.randint(
+            MINIMUM_ORDER_QUANTITY, MAXIMUM_ORDER_QUANTITY, size=(NUMBER_OF_SYMBOLS * 2,)
+        ),
+    )
+    for _ in range(1000)
+]
+configuration = fhe.Configuration(
+    enable_unsafe_features=True,
+    use_insecure_key_cache=True,
+    insecure_key_cache_location=".keys",
+    fhe_simulation=True,
+    comparison_strategy_preference=fhe.ComparisonStrategy.ONE_TLU_PROMOTED,
+    bitwise_strategy_preference=fhe.BitwiseStrategy.ONE_TLU_PROMOTED,
+    multivariate_strategy_preference=fhe.MultivariateStrategy.PROMOTED,
+    min_max_strategy_preference=fhe.MinMaxStrategy.ONE_TLU_PROMOTED,
```

Comment:
Did you try other min_max strategies?

Reply:
I will

Reply:
I had a look: with current

```
    comparison_strategy_preference=fhe.ComparisonStrategy.ONE_TLU_PROMOTED,
    bitwise_strategy_preference=fhe.BitwiseStrategy.ONE_TLU_PROMOTED,
    multivariate_strategy_preference=fhe.MultivariateStrategy.PROMOTED,
    min_max_strategy_preference=fhe.MinMaxStrategy.ONE_TLU_PROMOTED,
```

we have

```
types_differ = bank_order_types.reshape(-1, 1) != client_order_types

  %3 = "FHELinalg.apply_lookup_table"(%2, %cst) : (tensor<10x5x!FHE.esint<6>>, tensor<64xi64>)

symbols_match = bank_order_symbols.reshape(-1, 1) == client_order_symbols

  %7 = "FHELinalg.apply_lookup_table"(%6, %cst_1) : (tensor<10x5x!FHE.esint<6>>, tensor<64xi64>)

can_fill = symbols_match & types_differ

  %10 = "FHELinalg.apply_lookup_table"(%9, %cst_2) : (tensor<10x5x!FHE.eint<2>>, tensor<4xi64>)

matching_quantity = np.minimum(

  %14 = "FHELinalg.apply_lookup_table"(%13, %cst_4) : (tensor<10x5x!FHE.esint<7>>, tensor<128xi64>)

filled_quantity = fhe.multivariate(lambda x, y: x * y)(can_fill, matching_quantity)

  %18 = "FHELinalg.apply_lookup_table"(%17, %cst_6) : (tensor<10x5x!FHE.eint<7>>, tensor<128xi64>)
```

If I am not wrong, the first two are impacted by `comparison_strategy_preference`, the next one is impacted by `bitwise_strategy_preference`, the next one by `min_max_strategy_preference` and the final one by `multivariate_strategy_preference`

Reply:
To me, `bitwise_strategy_preference=fhe.BitwiseStrategy.ONE_TLU_PROMOTED,` will always be a good idea since we AND two bits, and the rest looks good for 50 values. 

I would keep it like this, confirmed @umut-sahin ?

Reply:
We can set it to default for 1b and 2b, maybe 3b, but after that it doesn't make sense at all (e.g., in 5b & 5b, it'll promote both operands to 10b)

Reply:
I see what you mean but I didn't intend to make something optimal for all sizes. I think it would be quite ambitious, and maybe take too much time for not so much to win, here.

Reply:
Closing since @umut-sahin agreed to merge.